### PR TITLE
Issue #23 - Add optional command to generate controllers, requests an…

### PR DESCRIPTION
**Pull request by b8ne**
Hey @tabacitu hope this does the trick.
Happy to change any names to suit conventions if needed.
Apart from that methods closely follow Backpack\Generator implementations as suggested.
Cheers

### CODE REVIEW

**Comment by tabacitu** [comment]
Omg! Thank you! :-) I'll take a closer look and merge it tonight or tomorrow. Cheers!

**Comment by b8ne** [comment]
Haha all good.
Just a heads up, ive used laravel a bit but this is my first project delving into packages.
So Happy to hear any advice.

**Comment by b8ne** [comment]
hey @tabacitu , have you had a chance to take a look at this one yet?

**Comment by tabacitu** [comment]
10 days ago?! Boy, times flies by. Sorry, I've been doublebooked these past weeks with client work and haven't had the time to give all PRs the attention they deserve.

But I do see you've put a lot of time into it, it looks really nice and clean and I really really appreciate it. I'll take a look this weekend and merge. Thank you!

**Comment by gpressutto5** [comment]
Boy, times flies by.

**Comment by b8ne** [comment]
@gpressutto5 yupppp :)

**Comment by b8ne** [comment]
Hey @tabacitu can you please have a look at merging this one? I havent been tracking updates so not sure if the files have changed much.  Would be good to get this in so I can switch my packages over and start receiving updates again.

**Comment by tabacitu** [comment]
Hi @b8ne ,

I'm so so sorry for not taking the time to merge this sooner. I'm not an asshole, I promise, it's just been hell managing everything and this is a great feature, but will definitely bring in more suggestions, bugs and things to manage. I was trying to minimize that. It's more of a "mature project" functionality, which, I'm glad to tell you, we're there :-) We're now FINALLY in the polish stage, where we make everything faster and better :-) 

Plus, I've been thinking about a separate package that does this sort of thing for everything - Backpack/CLI, which would allow you to:
- install backpack packages
- extend backpack packages
- generate models/controllers/requests
- publish routes

But I guess it could also be a collection of commands, each in its own package. At least to start with.

-----

I've taken it for a spin and here are a few things I would like to change. I've already done the modifs in the [extend-command branch](https://github.com/Laravel-Backpack/PermissionManager/commits/extend-command), just tell me what you think about them:
1. I'd change the command syntax a bit, to make it easier to remember:
- ```backpack:crud-permissions``` to ```backpack:extend:permission-manager```
- ```backpack:crud-permission-model``` to ```backpack:extend:permission-crud-model```
- ```backpack:crud-permission-request``` to ```backpack:extend:permission-crud-request```
- ```backpack:crud-permission-controller``` to ```backpack:extend:permission-crud-controller```
This way, any other extend command, from any other package, could follow the same syntax: ```backpack:extend:xyz```. And once they ALL have the extend command, you wouldn't even need to look up the command - it should look very similar to how you'd write it yourself.

2. I wouldn't publish the files inside a "Permissions" folder, but straight in the "Controllers\Admin", "Models" or "Requests" folder. What I'd want to achieve with this is to make it as intuitive as possible - by extending the PermissionManager the developer would basically have them very very similarly to how he would have generated them from scratch, using the "generate" command.

3. I'm not very sure about this, but I would name the published files either 
A) ```ExtendedPermissionCrudController``` 
or straight out 
B) ```PermissionCrudController```
I think I would prefer (B), actually. My judgement is on the same line - to make it just like the developer would have generated them himself. I'm just worried it might sometimes conflict with what's already there... But 99% of the time it shouldn't, right? 

Thanks, cheers!

**Comment by b8ne** [comment]
Sounds good mate :) All of your suggestions sound good.  I actually just found another bug regarding the request models and validation.  As you said, I think it may be best to step back, implement the better naming solutions and then have a look at it.

**Comment by lloy0076** [comment]
@tabacitu @b8ne - what's the status of this? Is this reviewable (as in should I spend time nit-picking)?

**Comment by JGamboa** [comment]
What happen with this?

**Comment by lloy0076** [comment]
@tabacitu @b8ne -- is there a status on this at all?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.idea/misc.xml
```
Comment (justRau) [inline .gitignore:1]: I think environment-specific files are better in a global .gitignore (https://gist.github.com/subfuzion/db7f57fff2fb6998a16c).

**Comment by tabacitu** [comment]
I've been thinking about this, and I don't think we should be making it any easier for people to customize their Controllers. If they need it, it isn't terribly difficult to do it manually. I think making this automated, and for all CRUDs, will promote a bad practice of overriding everything, and make this package unmaintainable.

Let's leave it manual.

Sorry it took me so long to reach this decision @b8ne . I know you probably don't care after all these years, maybe you're not even using Backpack any more, but I'm super-grateful for all the help 🙏 Sorry this didn't get merged.

Cheers!

**Pull request closed without merging**